src/datastore_mad/remotes/downloader.sh

```diff
@@ -295,7 +295,30 @@ function get_rbd_cmd
     echo "ssh '$(esc_sq "$DST_HOST")' \"$RBD export '$(esc_sq "$SOURCE")' -\""
 }
 
-TEMP=`getopt -o m:s:l:c:n -l md5:,sha1:,limit:,max-size:,convert:,nodecomp -- "$@"`
+# Compare 2 version strings using sort -V
+# Usage:
+#   verlte "3.2.9" "3.4.0"
+function verlte() {
+    [  "$1" = "`echo -e "$1\n$2" | sort -V | head -n1`" ]
+}
+
+# Returns curl retry options based on its version
+function curl_retry_args {
+    [ "$NO_RETRY" = "yes" ] && return
+
+    RETRY_ARGS="--retry 3 --retry-delay 3"
+
+    CURL_VER=`curl --version | grep -o 'curl [0-9\.]*' | awk '{print $2}'`
+
+    # To retry also on conn-reset-by-peer fresh curl is needed
+    if verlte "7.71.0" "$CURL_VER" && [ -z ${MAX_SIZE} ] ; then
+        RETRY_ARGS+=" --retry-all-errors"
+    fi
+
+    echo $RETRY_ARGS
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Double quote to prevent globbing and word splitting.